src/views/datatiles/data-tiles-directive.js

```diff
@@ -0,0 +1,339 @@
+            selectionMatchProp: 'uuid',
+            selectedItems: [],
+            checkDisabled: false,
+            showSelectBox: true,
+            onSelect: null,
+            onSelectionChange: null,
+            onCheckBoxChange: null,
+            onClick: null,
+            onDblClick: null
+          };
+
+          $scope.config = $.extend(true, angular.copy($scope.defaultConfig), $scope.config);
+          if ($scope.config.selectItems && $scope.config.showSelectBox) {
+            throw new Error('pfDataTiles - ' +
+            'Illegal use of pfDataTiles directive! ' +
+            'Cannot allow both select box and click selection in the same data tiles.');
+          }
+        }
+      ],
+
+      link: function (scope, element, attrs) {
+        attrs.$observe('config', function () {
+          scope.config = $.extend(true, angular.copy(scope.defaultConfig), scope.config);
+          if (!scope.config.selectItems) {
+            scope.config.selectedItems = [];
+          }
+          if (!scope.config.multiSelect && scope.config.selectedItems && scope.config.selectedItems.length > 0) {
+            scope.config.selectedItems = [scope.config.selectedItems[0]];
+          }
+        });
+
+        scope.itemClick = function (e, item) {
+          var alreadySelected;
+          var selectionChanged = false;
+          var continueEvent = true;
+
+          // Ignore disabled item clicks completely
+          if (scope.checkDisabled(item)) {
+            return continueEvent;
+          }
+
+          if (scope.config && scope.config.selectItems && item) {
+            if (scope.config.multiSelect && !scope.config.dblClick) {
+
+              alreadySelected = _.find(scope.config.selectedItems, function (itemObj) {
+                return itemObj === item;
+              });
+
+              if (alreadySelected) {
+                // already selected so deselect
+                scope.config.selectedItems = _.without(scope.config.selectedItems, item);
+              } else {
+                // add the item to the selected items
+                scope.config.selectedItems.push(item);
+                selectionChanged = true;
+              }
+            } else {
+              if (scope.config.selectedItems[0] === item) {
+                if (!scope.config.dblClick) {
+                  scope.config.selectedItems = [];
+                  selectionChanged = true;
+                }
+                continueEvent = false;
+              } else {
+                scope.config.selectedItems = [item];
+                selectionChanged = true;
+              }
+            }
+
+            if (selectionChanged && scope.config.onSelect) {
+              scope.config.onSelect(item, e);
+            }
+            if (selectionChanged && scope.config.onSelectionChange) {
+              scope.config.onSelectionChange(scope.config.selectedItems, e);
+            }
+          }
+          if (scope.config.onClick) {
```

Comment:
Not sure if it's worth saving the lines but you could default all of these event handlers to `function () {}` and then just call them.  I suppose the check is safer but, then again, it doesn't check that they are actually callable.